pkg/driver/screen/x11_linux.go

```diff
@@ -0,0 +1,107 @@
+package screen
+
+import (
+	"fmt"
+	"image"
+	"image/color"
+	"time"
+
+	"github.com/pion/mediadevices/pkg/driver"
+	"github.com/pion/mediadevices/pkg/frame"
+	"github.com/pion/mediadevices/pkg/io/video"
+	"github.com/pion/mediadevices/pkg/prop"
+)
+
+type screen struct {
+	num    int
+	reader *reader
+	tick   *time.Ticker
+}
+
+func deviceID(num int) string {
+	return fmt.Sprintf("X11Screen%d", num)
+}
+
+func init() {
+	dp, err := openDisplay()
+	if err != nil {
+		// No x11 display available.
+		return
+	}
+	defer dp.Close()
+	numScreen := dp.NumScreen()
+	for i := 0; i < numScreen; i++ {
+		driver.GetManager().Register(
+			&screen{
+				num: i,
+			},
+			deviceID(i),
+		)
+	}
+}
+
+func (s *screen) Open() error {
+	r, err := newReader(s.num)
+	if err != nil {
+		return err
+	}
+	s.reader = r
+	return nil
+}
+
+func (s *screen) Close() error {
+	s.reader.Close()
+	if s.tick != nil {
+		s.tick.Stop()
+	}
+	return nil
+}
+
+func (s *screen) VideoRecord(p prop.Media) (video.Reader, error) {
+	rect := s.reader.img.Bounds()
+	w := rect.Max.X - rect.Min.X
+	h := rect.Max.Y - rect.Min.Y
+	imgI444 := image.NewYCbCr(rect, image.YCbCrSubsampleRatio444)
+
+	if p.FrameRate == 0 {
+		p.FrameRate = 10
+	}
+	s.tick = time.NewTicker(time.Duration(float32(time.Second) / p.FrameRate))
+
+	r := video.ReaderFunc(func() (image.Image, error) {
+		<-s.tick.C
+		img := s.reader.Read()
+		// Convert it to I444
+		for y := 0; y < h; y++ {
+			iyBase := y * imgI444.YStride
+			icBase := y * imgI444.CStride
+			for x := 0; x < w; x++ {
+				iy := iyBase + x
+				ic := icBase + x
+				r, g, b, _ := img.At(x, y).RGBA()
+				yy, cb, cr := color.RGBToYCbCr(uint8(r/0x100), uint8(g/0x100), uint8(b/0x100))
+				imgI444.Y[iy] = yy
+				imgI444.Cb[ic] = cb
+				imgI444.Cr[ic] = cr
+			}
+		}
+		return imgI444, nil
+	})
+	return r, nil
+}
+
+func (s *screen) Properties() []prop.Media {
+	rect := s.reader.img.Bounds()
+	w := rect.Max.X - rect.Min.X
+	h := rect.Max.Y - rect.Min.Y
+	return []prop.Media{
```

Comment:
It looks like we need extra information to let us know that an adapter can be used for screen sharing. I saw that there's a web API called, https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/getDisplayMedia. It's very similar to `getMediaDevices` but for screen sharing devices.

I've created a POC branch that implements `getDisplayMedia`, https://github.com/pion/mediadevices/tree/GetDisplayMedia. Please let me know what you think about it.

POC Overview:

    Change driver registration requirement to have an Info
    
    * Move label to Info
    * Replace label requirement to be an info
    * Replace Label driver method to be an Info method

    Add DeviceType to Info and a filter function for it
    
    * Add DeviceType to Info
    * Update adapters to also register its device type
    * Add FilterDeviceType

    Add GetDisplayMedia
    
    * Add selectScreen to select all registered drivers that can screenshare
    * Add GetDisplayMedia
    * Add screenshare example

Reply:
Oh, I didn't know `getDisplayMedia`.

On browser, `enumerateDevice` seems not listing screen capture devices, and I guess `getUserMedia` never select screen capture device.
Maybe it's better to omit screen capture from our `GetDisplayMedia`?
(But, enumerating screens is helpful.)

Please feel free to push to this branch.

Reply:
Hmm, that's interesting. I agree I think we should omit screen capture devices from `getUserMedia`. 

btw, are you suggesting that we should still keep the screen capture devices for `EnumerateDevices`?

Reply:
> btw, are you suggesting that we should still keep the screen capture devices for EnumerateDevices?

I'm not very sure it's good, but there may be multiple screens the user can select.
At least, there should be a way to get the number of the screens.

Reply:
Sadly, the spec for `enumerateDevices` doesn't say anything about screen capture devices, it only talks about camera and microphone. https://w3c.github.io/mediacapture-main/#dom-mediadevices-enumeratedevices.

Reply:
> > btw, are you suggesting that we should still keep the screen capture devices for EnumerateDevices?
> 
> I'm not very sure it's good, but there may be multiple screens the user can select.
> At least, there should be a way to get the number of the screens.

That's definitely a good point. Should we just expose `DeviceType` in `EnumerateDevices`. it's kinda weird that it doesn't follow the spec. But, I think it's pretty clean. What do you think?

Reply:
> Should we just expose DeviceType in EnumerateDevices. 

Totally agree with this!

Reply:
Sweet. I'll add this then.

Reply:
added